### PR TITLE
EOS-19276 - SSPL RPM installation failure - %pre exit with return code 1 and fixing permission issue

### PR DESCRIPTION
**Pull request by mariyappanp**

### CODE REVIEW

File: low-level/sspl-ll.spec [hunk 1]
```diff
@@ -78,7 +78,9 @@ cp -p $SSPL_SETUP/consuldump.py $SSPL_BASE/low-level/
 # take backup of cache folder if exists
 mkdir -p /opt/seagate/backup/%{version}
```
Comment (cdeshmukh) [inline low-level/sspl-ll.spec:79]: This backup folder should stay with in /opt/seagate/cortx

Also backup should limit to /var/%{product_family}/sspl/**data**, target folder being /opt/seagate/cortx/sspl/data, if its meant to retain just persistence folder & sspl.conf too, why sspl folder?

File: low-level/sspl-ll.spec [hunk 2]
```diff
@@ -78,7 +78,9 @@ cp -p $SSPL_SETUP/consuldump.py $SSPL_BASE/low-level/
...
 [ -f /etc/sspl.conf ] && cp -p /etc/sspl.conf /opt/seagate/backup/%{version}/sspl.conf
-[ -d /var/%{product_family}/sspl ] && cp -Rp /var/%{product_family}/sspl /opt/seagate/backup/%{version}/
+if [ -d /var/%{product_family}/sspl ]; then
```
Comment (cdeshmukh) [inline low-level/sspl-ll.spec:81]: logic is same as existing code, what different is being achieved here? Why this is failing installation suddenly ?

Comment (mariyappanp) [reply low-level/sspl-ll.spec:81]: If condition won't be satisfied, it gives rc 1 for single line bash.

> [root@ssc-vm-3120 ~]# [ -d /var/cortx/sspl ] && echo "hi"
> [root@ssc-vm-3120 ~]#
> [root@ssc-vm-3120 ~]# echo $?
> 1

Same condition but this gives rc 0.

> [root@ssc-vm-3120 ~]# if [ -d /var/cortx/sspl ]; then
> > echo "hi"
> > fi
> [root@ssc-vm-3120 ~]#
> [root@ssc-vm-3120 ~]# echo $?
> 0

File: low-level/sspl-ll.spec (hunk 1 above)
Comment (mariyappanp) [reply low-level/sspl-ll.spec:79]: This is legacy. But I agree with your point.
I will bring the changes.

Comment (mariyappanp) [reply low-level/sspl-ll.spec]: Done.

> [root@ssc-vm-3120 ~]# ls -lrt /opt/seagate/cortx/backup/2.0.0/sspl/
> total 12
> drwxr-xr-x 5 sspl-ll sspl-ll 4096 Apr  6 13:24 data
> -rw-r--r-- 1 root    root    4176 Apr  6 13:34 sspl.conf

File: low-level/sspl-ll.spec [hunk 3]
```diff
@@ -76,9 +76,11 @@ cp -p $SSPL_SETUP/consuldump.py $SSPL_BASE/low-level/
...
 %pre
 # take backup of cache folder if exists
-mkdir -p /opt/seagate/backup/%{version}
-[ -f /etc/sspl.conf ] && cp -p /etc/sspl.conf /opt/seagate/backup/%{version}/sspl.conf
-[ -d /var/%{product_family}/sspl ] && cp -Rp /var/%{product_family}/sspl /opt/seagate/backup/%{version}/
+mkdir -p /opt/seagate/%{product_family}/backup/%{version}/sspl
+[ -f /etc/sspl.conf ] && cp -p /etc/sspl.conf /opt/seagate/%{product_family}/backup/%{version}/sspl/sspl.conf
+if [ -d /var/%{product_family}/sspl/data ]; then
+    cp -Rp /var/%{product_family}/sspl/data /opt/seagate/%{product_family}/backup/%{version}/sspl/
+fi
```
Comment (sandeepsgt) [inline low-level/sspl-ll.spec:83]: Why this is failing now, the same code used to work earlier

Comment (mariyappanp) [reply low-level/sspl-ll.spec:83]: Earlier we had directory creation after this condition. rc was still same at this condition but after the directory creation execution rc was 0.
So we hadn't faced this issue.

**Review by cdeshmukh — APPROVED** [review]

**Review by sandeepsgt — APPROVED** [review]

**Pull request merged**